### PR TITLE
[Work in Progress] 2.0 Documentation

### PR DESCRIPTION
**Pull request by jbaxleyiii**
This PR looks to vastly improve the state of documentation for how to build and run a GraphQL server.

### CODE REVIEW

File: docs/_config.yml [hunk 1]
```diff
@@ -7,32 +7,45 @@ versions:
...
+    - schemas/context
+    - schemas/organizing-code
+  Running a Server:
+    - server/index
+    - server/middleware
+    - server/queries
+    - server/engine
+    - server/secrets
+  Best Practices:
+    - best-practices/authentication
```
Comment (stubailo) [inline docs/_config.yml]: Maybe I'll do this one

File: docs/_config.yml [hunk 2]
```diff
@@ -7,32 +7,45 @@ versions:
...
+    - best-practices/schema-design
```
Comment (stubailo) [inline docs/_config.yml]: I'll take a stab at this

File: docs/_config.yml [hunk 3]
```diff
@@ -7,32 +7,45 @@ versions:
...
+    - best-practices/subscriptions
+    - best-practices/performance
+    - best-practices/caching
+    - best-practices/monitoring
+    - best-practices/versioning
+    - best-practices/schema-stitching
```
Comment (stubailo) [inline docs/_config.yml]: Let's move this to backends

File: docs/_config.yml (hunk 2 above)
Comment (evans) [reply docs/_config.yml]: I think making every type nullable should be a best practice, unless an instance of the type cannot exist without the non-null field: https://medium.com/@calebmer/when-to-use-graphql-non-null-fields-4059337f6fc8

File: docs/source/advanced/query-mechanics.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+title: Query mechanics
```
Comment (stubailo) [inline docs/source/advanced/query-mechanics.md]: Do we need this article in the first version?

File: docs/source/best-practices/authentication.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+title: Authentication and Authorization
```
Comment (stubailo) [inline docs/source/best-practices/authentication.md]: Should authorization be in here? Also, why mutations specifically? Maybe it's too early to review this article ATM

File: docs/source/best-practices/schema-stitching.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+---
+title: Schema Stitching
```
Comment (stubailo) [inline docs/source/best-practices/schema-stitching.md]: Let's kill this article for now - we don't know what the best practices are here.

File: docs/source/server/queries.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    process: Process
+    argv: [String]
+  }
+`;
+
+// Resolvers define the technique for fetching the types in the
+// schema.  We'll retrieve books from the "books" array above.
+const resolvers = {
+  Process: {
+    params: (parent) => parent.argv.slice(1)
```
Comment (stubailo) [inline docs/source/server/queries.md]: How did we end up with this `process` example? This is a bit meta IMO

File: docs/source/why-apollo.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
+---
+title: Why Apollo?
```
Comment (stubailo) [inline docs/source/why-apollo.md]: Why Apollo Server

File: docs/source/advanced/query-mechanics.md (hunk 4 above)
Comment (abernix) [reply docs/source/advanced/query-mechanics.md]: This is a lone-wolf in `advanced` right now too.  My comment with the same perspective: https://github.com/apollographql/apollo-server/pull/942#discussion_r182720602

File: docs/source/best-practices/authentication.md (hunk 5 above)
Comment (evans) [reply docs/source/best-practices/authentication.md]: I think authorization fits in nicely for now. If we expand on the different ways to authenticate, then we might split it up

**Comment by abernix** [comment]
Thanks to everyone who helped contribute to this documentation work!

We certainly aren't done with improving it, but it's now available for the new Apollo Server 2.0 **beta** at https://www.apollographql.com/docs/apollo-server/v2/.

**Pull request merged**